src/main/java/de/tudarmstadt/lt/cw/CW.java

```diff
@@ -75,8 +96,28 @@ protected void relabelNode(N node) {
 			if (edge == null) {
 				break;
 			}
-			N label = nodeLabels.get(edge.getSource());
-			MapUtil.addFloatTo(labelScores, label, edge.getWeight());
+			N sourceNode = edge.getSource();
+			N label = nodeLabels.get(sourceNode);
+			Float weight = edge.getWeight();
+			switch (getOption()) {
+			case DIST_LOG: {
+				float neighbours = count(graph.getNeighbors(sourceNode));
+				weight = (float) (weight / (Math.log(neighbours + 1)));
+				break;
+			}
+			case DIST_NOLOG: {
+				float neightbours = count(graph.getNeighbors(sourceNode));
+				weight = weight / neightbours;
+				break;
+			}
+			case TOP:
+				// nothing to do, use the weight of the edge
```

Comment:
Just a matter of style, but I would rather call edge.getWeight() every time you need weight above (and here you would do also weight = edge.getWeight(). In this case, you do not need such a comment.